parser_library/src/workspaces/processor_file_impl.cpp

```diff
@@ -53,12 +53,13 @@ parse_result processor_file_impl::parse(parse_lib_provider& lib_provider,
     if (!last_analyzer_opencode_)
         last_opencode_id_storage_ = std::make_shared<context::id_storage>();
 
+    const bool collect_hl = should_collect_hl();
     auto new_analyzer = std::make_unique<analyzer>(get_text(),
         analyzer_options {
             get_location(),
             &lib_provider,
             std::move(asm_opts),
-            get_lsp_editing() ? collect_highlighting_info::yes : collect_highlighting_info::no,
+            collect_hl ? collect_highlighting_info::yes : collect_highlighting_info::no,
```

Comment:
```suggestion
            should_collect_hl()? collect_highlighting_info::yes : collect_highlighting_info::no,
```

To unify with other calls?

Reply:
I need the value later on in the function.

Reply:
Right...